src/OpenTelemetry.Api/Internal/ActivityHelperExtensions.cs

```diff
@@ -109,14 +109,15 @@ public static bool TryCheckFirstTag(this Activity activity, string tagName, out
         /// <typeparam name="T">The struct <see cref="IActivityEnumerator{T}"/> implementation to use for the enumeration.</typeparam>
         /// <param name="activity">Activity instance.</param>
         /// <param name="tagEnumerator">Tag enumerator.</param>
+        /// <param name="maxTags">Maximum number of tags to enumerate.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "ActivityProcessor is hot path")]
-        public static void EnumerateTags<T>(this Activity activity, ref T tagEnumerator)
+        public static void EnumerateTags<T>(this Activity activity, ref T tagEnumerator, int? maxTags = null)
```

Comment:
Would it better to add an overload which takes in an additional non-nullable `int maxTags` as a parameter?

Reply:
I guess you'll also have to set the default values for these parameters these config variables then.

Reply:
Yes? 😄 

I'm usually not a fan of optional parameters - especially for public APIs, but since this is internal I was a bit lazy.

Curious what your thoughts are. Do you similarly not prefer optional parameters, or is there another reason you think a proper overload would be better in this case? I'm fine changing this either way.

Reply:
I was suggesting that because that wat we could get rid of the extra `if` check for regular scenarios. 😄

Reply:
Oh yea, I see. Since a lot of this code will change or go away if we drop DiagnosticSource 6, I'm inclined to just leave it as is for the moment. What do you think?

Reply:
Yeah, I think that should be fine.